src/musicbrainz/web/coverartarchivelinkstask.cpp

```diff
@@ -0,0 +1,157 @@
+    DEBUG_ASSERT(url.isValid());
+    QNetworkRequest networkRequest(url);
+    return networkRequest;
+}
+
+} // anonymous namespace
+
+CoverArtArchiveLinksTask::CoverArtArchiveLinksTask(
+        QNetworkAccessManager* pNetworkAccessManager,
+        const QUuid& albumReleaseId,
+        QObject* pParent)
+        : network::JsonWebTask(
+                  pNetworkAccessManager,
+                  kBaseUrl,
+                  lookupRequest(),
+                  pParent),
+          m_albumReleaseId(albumReleaseId) {
+}
+
+QNetworkReply* CoverArtArchiveLinksTask::sendNetworkRequest(
+        QNetworkAccessManager* pNetworkAccessManager,
+        network::HttpRequestMethod method,
+        const QUrl& url,
+        const QJsonDocument& content) {
+    DEBUG_ASSERT(pNetworkAccessManager);
+    Q_UNUSED(method);
+    DEBUG_ASSERT(method == network::HttpRequestMethod::Get);
+    pNetworkAccessManager->setRedirectPolicy(QNetworkRequest::NoLessSafeRedirectPolicy);
+
+    const QNetworkRequest networkRequest = createNetworkRequest(m_albumReleaseId);
+
+    VERIFY_OR_DEBUG_ASSERT(url.isValid()) {
+        kLogger.warning() << "Invalid URL" << url;
+        return nullptr;
+    }
+    if (kLogger.traceEnabled()) {
+        kLogger.trace()
+                << "Get"
+                << url;
+    }
+    return pNetworkAccessManager->get(networkRequest);
+}
+
+void CoverArtArchiveLinksTask::onFinished(
+        const network::JsonWebResponse& response) {
+    if (!response.isStatusCodeSuccess()) {
+        kLogger.warning()
+                << "Request failed with HTTP status code"
+                << response.statusCode();
+        emitFailed(response);
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(response.statusCode() == network::kHttpStatusCodeOk) {
+        kLogger.warning()
+                << "Unexpected HTTP status code"
+                << response.statusCode();
+        emitFailed(response);
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(response.content().isObject()) {
+        kLogger.warning()
+                << "Invalid JSON content"
+                << response.content();
+        emitFailed(response);
+        return;
+    }
+    const auto jsonObject = response.content().object();
+
+    if (jsonObject.isEmpty()) {
+        kLogger.warning()
+                << "Empty Json";
+        emitFailed(response);
+        return;
+    }
+
+    QList<QString> allUrls;
+    DEBUG_ASSERT(jsonObject.value(QLatin1String("images")).isArray());
+    const QJsonArray images = jsonObject.value(QLatin1String("images")).toArray();
+    for (const auto& image : images) {
+        DEBUG_ASSERT(image.isObject());
+        const auto imageObject = image.toObject();
+
+        const auto thumbnails = imageObject.value(QLatin1String("thumbnails")).toObject();
+        DEBUG_ASSERT(!thumbnails.isEmpty());
+
+        const auto smallThumbnailUrl = thumbnails.value(QLatin1String("small")).toString();
```

Comment:
This might fail and hit the assertions for the last tag in Abba - Super Trouper. 

The link of the release is: https://coverartarchive.org/release/5240094f-9d79-44fd-9985-77c7287bcc16

As far as I tested, I've only found this release as a different response than the others. Maybe there are others that might don't have small-large and have 250-500, I'm not sure.

Should we cover both "250" and "small" if small doesn't exists then try 250 or vice versa?

Reply:
No, than don't change the code.
Just add a comment about our decision.

Reply:
Done, thank you!